packages/pluggableWidgets/app-events-native/src/AppEvents.tsx

```diff
@@ -95,8 +100,109 @@ export class AppEvents extends Component<Props> {
 
         this.isConnected = nextState.isConnected;
     }
+
+    private triggerOnUnloadEvent(): void {
+        if (this.props.onUnloadAction && this.props.onUnloadAction.canExecute) {
+            executeAction(this.props.onUnloadAction);
+        }
+    }
+
+    private triggerOnLoadEvent(): void {
+        if (!this.onLoadTriggered && this.props.onLoadAction?.canExecute) {
+            this.onLoadTriggered = true;
+            executeAction(this.props.onLoadAction);
+        }
+    }
+
+    private registerScheduledExecution(): void {
+        if (this.props.onTimeoutAction) {
+            this.onTimeoutAction = scheduleTask(
+                () => executeAction(this.props.onTimeoutAction),
+                this.props.delayTime * 1000,
+                this.props.timerType
+            );
+        }
+    }
 }
 
 function isPastTimeout(last: number, timeoutSeconds: number): boolean {
     return Date.now() - last >= timeoutSeconds * 1000;
 }
+
+interface ScheduledExecution {
+    cancel: () => boolean;
+    schedule: () => boolean;
+}
+
+class ScheduledOnceExecution {
+    private timeoutHandle: number | undefined;
+    private _finished = false;
+    private _running = false;
+
+    constructor(private readonly fn: () => void, private readonly interval: number) {
+        this.schedule();
+    }
+
+    cancel(): boolean {
```

Comment:
Do we have a reason why we are returning boolean's on `cancel` and `schedule` functions ? I dont see them being used 🤷

Reply:
api clarity

Reply:
removed